targets/TARGET_NORDIC/TARGET_NRF5x/nordic_critical.c

```diff
@@ -53,6 +53,13 @@ void core_util_critical_section_exit()
 #endif
 }
 
+
+bool hal_in_critical_section(void)
```

Comment:
Should this be surrounded by `#if defined(SOFTDEVICE_PRESENT)`? The state_saved variable is only defined if this macros is true. Compilation will fail without the macro.

Reply:
Or should the variable be defined outside the `#ifdef`?

Reply:
Man....
It really does look like it should. @marcuschangarm?

Reply:
Talked with @marcuschangarm. It's not an immediate worry since it's apparently _really_ hard to configure mbed such that this code path is activated, but he's also doing more code refactoring to where this shouldn't be an issue.